.github/workflows/on-pull-request.yml

```diff
@@ -76,6 +81,40 @@ jobs:
         with:
           eslint: true
 
+  megalinter:
+    name: MegaLinter
+    runs-on: ubuntu-latest
+    steps:
+      # Git Checkout
+      - name: Checkout Code
+        uses: actions/checkout@v3
+        with:
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      # MegaLinter
+      - name: MegaLinter
+        id: ml
+        # You can override MegaLinter flavor used to have faster performances
+        # More info at https://megalinter.io/flavors/
+        uses: oxsecurity/megalinter/flavors/javascript@v6
+        env:
+          # All available variables are described in documentation
+          # https://megalinter.io/configuration/
+          APPLY_FIXES: all
+          VALIDATE_ALL_CODEBASE: true # Set ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }} to validate only diff with main branch
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          # ADD YOUR CUSTOM ENV VARIABLES HERE TO OVERRIDE VALUES OF .mega-linter.yml AT THE ROOT OF YOUR REPOSITORY
+
+      # Upload MegaLinter artifacts
+      - name: Archive production artifacts
+        if: ${{ success() }} || ${{ failure() }}
+        uses: actions/upload-artifact@v3
+        with:
+          name: MegaLinter reports
+          path: |
+            megalinter-reports
+            mega-linter.log
+
```

Comment:
```suggestion
     - uses: actions/upload-artifact@v3
        with:
          name: megalinter-reports
          path: |
            megalinter-reports
            mega-linter.log

```

Reply:
I'm pretty sure you need this one, else the default condition is `success()`
But you obviously want to upload reports when MegaLinter fails :)

Reply:
Then we could write it this way?

if: always()

Reply:
I think it did not exist at the time MegaLinter was created, but it's worth a try :)
If it works I'll even change the default ML workflow ^^